src/parser.rs

```diff
@@ -357,6 +303,57 @@ fn simd_unhex(value: &[u8]) -> u32 {
     shifted.reduce_or()
 }
 
+const SIMD_SPACE_CHAR: Simd<u8, 32> = u8x32::from_array([b' '; 32]);
+const SIMD_NEWLINE_CHAR: Simd<u8, 32> = u8x32::from_array([b'\n'; 32]);
+const SIMD_0_CHAR: Simd<u8, 32> = u8x32::from_array([b'0'; 32]);
+const SHUFFLE_PATTERNS: [(u8, Simd<u8, 32>); u16::MAX as usize + 1] =
+    manually_calculate_shuffle_patterns();
+const DECIMAL_FACTORS_X: Simd<u16, 16> =
+    u16x16::from_array([1000, 100, 10, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+const DECIMAL_FACTORS_Y: Simd<u16, 16> =
+    u16x16::from_array([0, 0, 0, 0, 1000, 100, 10, 1, 0, 0, 0, 0, 0, 0, 0, 0]);
+
+// Longest possible space bitmask = "1234 1234 " => 10 chars
+const SPACES_BITMASK_MASK: u32 = 0b0000_0000_0000_0000_0011_1111_1111;
+
+// Input: 32 characters starting where the x coordinate starts, eg. "1234 4321<random chars follow>"
+// Returns: (x, y, total length of text containing "<x> <y>")
+#[inline(always)]
+fn simd_parse_coord(value: &[u8]) -> (u16, u16, u8) {
+    #[cfg(debug_assertions)]
+    assert!(value.len() >= 10);
+
+    // let chars = unsafe { u8x32::from_array(*value.as_ptr().cast()) };
+    let chars = u8x32::from_array([
+        value[0], value[1], value[2], value[3], value[4], value[5], value[6], value[7], value[8],
+        value[9], 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    ]);
+    // dbg!(chars);
+
+    let digits = chars - SIMD_0_CHAR;
+
+    // ATTENTION: Bitmask starts with LSB (so kind of wrong order)
+    let spaces_bitmask = chars.simd_eq(SIMD_SPACE_CHAR).to_bitmask();
+    let newline_bitmask = chars.simd_eq(SIMD_NEWLINE_CHAR).to_bitmask();
+    let spaces_bitmask = (spaces_bitmask | newline_bitmask) & SPACES_BITMASK_MASK;
```

Comment:
I'm just doing a `digits>=10` on digits, as everything below `b'0'` will wrap around, this catches all characters other than digits.